File: backend/RentingServiceBackend/Models/PostDto.cs
using RentingServiceBackend.Entities;

namespace RentingServiceBackend.Models
{
    public class PostDto
    {
        public int PostId { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Image { get; set; }
        public string Location { get; set; }
        public DateTime AddDate { get; set; }
        public int FollowCount { get; set; }
        public List<string> Features { get; set; }
        public List<string> Categories { get; set; }
        public List<CommentDto> Comments { get; set; }
        public int Rate { get; set; }
        public UserDto User { get; set; }
    }
}

File: frontend2/RentingServiceFrontend/src/app/interfaces/postDto.ts
import { commentDto } from './commentDto';
import { userDto } from './userDto';

export interface postDto {
  postId: number;
  title: string;
  description: string;
  mainCategory: string;
  squareFootage: number;
  price: number;
  picturesPath: string;
  pictures: string[];
  lat: string;
  lng: string;
  buildingNumber: string;
  street: string;
  district: string;
  city: string;
  country: string;
  addDate: Date;
  followCount: number;
  isFollowedByUser: boolean | null;
  user: userDto;
  confirmed: boolean;
  sleepingPlaceCount: number | null;
  features: string[];
  comments: commentDto[];
  rate: number | null;
}
